src/psyclone/domain/lfric/kernel/array_arg_metadata.py

```diff
@@ -0,0 +1,159 @@
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author L. Turner, Met Office
+
+'''Module containing the ArrayArgMetadata class which captures the metadata
+associated with an array argument. Supports the creation, modification
+and Fortran output of an Array argument.
+
+'''
+from psyclone.domain.lfric import LFRicConstants
+from psyclone.domain.lfric.kernel.scalar_arg_metadata import ScalarArgMetadata
+
+
+class ArrayArgMetadata(ScalarArgMetadata):
+    '''Class to capture LFRic kernel metadata information for an array
+    argument.
+
+    :param str datatype: the datatype of this array (GH_INTEGER, ...).
+    :param str access: the way the kernel accesses this array (GH_READ).
+    :param str function_space: the function space that this array is \
+        on (W0, ...).
+
+    '''
+    # The name used to specify an array argument in LFRic metadata.
+    form = "gh_array"
+    # The relative positions of LFRic metadata. Metadata for an array
+    # argument is provided in the following format 'arg_type(form,
+    # datatype, access, function_space)'. Therefore, for example, the
+    # index of the form argument (form_arg_index) is 0.
+    form_arg_index = 0
+    datatype_arg_index = 1
+    access_arg_index = 2
+    array_size_arg_index = 3
+    # The name to use for any exceptions.
+    check_name = "array"
+    # The number of arguments in the language-level metadata (min and
+    # max values).
+    nargs = (4)
+
+    def __init__(self, datatype, access, array_ndims):
+        super().__init__(datatype, access)
+        self.array_ndims = array_ndims
+
+    @classmethod
+    def _get_metadata(cls, fparser2_tree):
+        '''Extract the required metadata from the fparser2 tree and return it
+        as strings. Also check that the metadata is in the expected
+        form (but do not check the metadata values as that is done
+        separately).
+
+        :param fparser2_tree: fparser2 tree containing the metadata
+            for this argument.
+        :type fparser2_tree: :py:class:`fparser.two.Fortran2003.Part_Ref` |
+            :py:class:`fparser.two.Fortran2003.Structure_Constructor`
+
+        :returns: a tuple containing the datatype, access and array ndims
+            metadata.
+        :rtype: Tuple[str, str, str]
+
+        '''
+        datatype, access = super()._get_metadata(fparser2_tree)
+        array_ndims = cls.get_array_ndims(fparser2_tree)
+        return (datatype, access, array_ndims)
+
+    def fortran_string(self):
+        '''
+        :returns: the metadata represented by this class as Fortran.
+        :rtype: str
+        '''
+        return (f"arg_type({self.form}, {self.datatype}, {self.access}, "
+                f"{self.array_ndims})")
+
+    @staticmethod
```

Comment:
Unless I'm missing something, this whole method can be removed as it's identical to the one in the class from which this one inherits.